modules/core/src/main/java/org/apache/ignite/internal/util/nio/GridNioSessionImpl.java

```diff
@@ -269,6 +273,22 @@ public void reset0() {
         return accepted;
     }
 
+    /** */
+    @Override public Certificate[] certificates() {
+        GridSslMeta meta = meta(SSL_META.ordinal());
+
+        if (meta != null) {
+            try {
+                return meta.sslEngine().getSession().getPeerCertificates();
+            }
+            catch (SSLPeerUnverifiedException e) {
```

Comment:
Not sure, but we probably want to pass that exception along, not swallow it. Let the caller deal with it.

Reply:
This is expected exception - can be thrown in perfectly legitimate scenarios.